### DISCUSSION TITLE
Will validation_step use torch.no_grad?

### DISCUSSION THREAD

**Discussion by morestart** [Lightning-AI/pytorch-lightning#9728] [General]
I want know will validation_step use torch.no_grad to eval model or just use model.eval()?

**Comment by ananthsub** [comment]
yes, no_grad is automatically applied around the validation loop: https://github.com/PyTorchLightning/pytorch-lightning/blob/131176b9f5e544a1133240a46dca1960308b7849/pytorch_lightning/trainer/trainer.py#L1195-L1196

https://github.com/PyTorchLightning/pytorch-lightning/blob/131176b9f5e544a1133240a46dca1960308b7849/pytorch_lightning/loops/epoch/training_epoch_loop.py#L254-L255

**Reply by morestart** [reply]
oh, thanks for your help~
